### DISCUSSION TITLE
I get a lot of 502 since yesterday

### DISCUSSION THREAD

**Discussion by Mihai-B** [trakt/trakt-api#339] [Q&A] [answered]
For some reason, when a user's downloads the watch history for a particular show, Trakt sends back a 502. 

Is it only me that has this issue or is it a general problem ?

Edit: Checking the logs, I can see that different users with different devices have this problem. and it seems that they get 502 on different endpoints. some receive it on episode summary, some on watch progress, and some on the user's last activity endpoint

**Comment by eliehaykal** [comment]
Yes, there seems to be an issue with a few endpoints. I am also receiving these errors.

**Comment by UweTrottmann** [comment]
Can confirm. Started 2022-04-13, affects various API calls, at least 1800 affected installations/users over the last two days. Not seeing it myself (Germany). So likely regional?

**Comment by OptimusGREEN** [comment]
Same issue here in the uk

**Comment by maxhand** [comment]
I've also been experiencing similar errors on my end since yesterday afternoon. 

On a related note, is there an uptime status page available? Such a page would be useful in situations like this.

Thanks!

**Reply by UweTrottmann** [reply]
https://status.trakt.tv/

**Reply by maxhand** [reply]
Seems useful, except it doesn't appear to be indicating any current issues?

**Comment by UweTrottmann** [comment]
> We’re looking into the API slowness and 502 errors. Will post more details once we know what’s going on.

https://twitter.com/traktapi/status/1514602500090826762

**Reply by maxhand** [reply]
Thank you Uwe!

**Reply by Mihai-B** [reply]
The problem is, it's not api slownes, it just sends 502 out of the blue. It goes fine and then all of the suden you receive a 502

**Comment by Mihai-B** [comment]
It looks like it is back up

**Comment by UweTrottmann** [comment, marked as answer]
> API performance is improving. We added a bunch more servers to help stabilize things and are continuing to investigate the cause.
>
> App developers: please optimize your apps and cache data locally as much as possible. This helps all apps that use the API.

https://twitter.com/traktapi/status/1514682311337394178

No more 502 reports since around 2022-04-14 6:00pm UTC in my case.

**Comment by maxhand** [comment]
It was very stable yesterday evening (EST), but as of this morning the slowness seems to have returned, and I'm getting occasional 502's.
